tools/class-redirector/class-redirector/ClassRedirectorTask.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+#nullable enable
+
+namespace ClassRedirector
+{
+	public class ClassRedirectorTask : Microsoft.Build.Utilities.Task
+	{
+		public override bool Execute ()
```

Comment:
There's no need to compute the xamarin assembly name, because we already know it and we can pass it as a property:

```csharp
[Required]
public string PlatformAssembly { get; set; } = string.Empty;
```

and then when calling the task from the targets:

```xml
<ClassRedirectorTask
		PlatformAssembly="$(_PlatformAssemblyName)"
		/>
```